inc/utils/content.php

```diff
@@ -444,37 +444,45 @@ function echo_post_meta($the_post, $show_elements = array('date','categories','t
 
 }
 
-function odm_excerpt($the_post, $num = 400, $read_more = '')
+function odm_excerpt($the_post, $num = 40, $read_more = '')
  {
 	  global $post;
 		$post = $the_post;
 		$limit = $num;
-		$chopped = false;
-		
-		if (odm_language_manager()->get_current_language() == "km"):
-			$limit = $limit * 4;
+
+		if($post->post_excerpt):
+			$post_content = $post->post_excerpt;
+		else:
+			$post_content = strip_tags(apply_filters('the_content',$post->post_content));
 		endif;
 
-		$post_content = apply_filters('the_content',$post->post_content);
-		$translated_content = apply_filters('translate_text',$post_content, odm_language_manager()->get_current_language());
+		$translated_content = apply_filters('translate_text', $post_content, odm_language_manager()->get_current_language());
 		$stripped_content = strip_tags($translated_content);
 		$stripped_content = strip_shortcodes($stripped_content);
 
-		$excerpt_hidden_space = explode("​", $stripped_content); //explode by zerowidthspace​
-		$excerpt_string = implode("​", $excerpt_hidden_space); //implode by zerowidthspace
-		$excerpt_words = $excerpt_string;
+		if($stripped_content){
+			$stripped_content = preg_replace( "/\&hellip;/",'', trim($stripped_content));
+			$stripped_content_arr = explode(' ', trim($stripped_content), $num+1);
 
-		if (strlen($excerpt_words) > $limit):
-			$excerpt_words = substr($excerpt_words,0,$limit);
-			$excerpt_words .= "...";
-		endif;
+			if($stripped_content_arr):
+				array_splice($stripped_content_arr, $limit);
+				$excerpt_content = implode(' ', $stripped_content_arr);
+		    if (odm_language_manager()->get_current_language() == "km"):
+		  		$excerpt_zeo_space = explode("​", $excerpt_content, $limit); //explode by zerowidthspace​
+		  		$excerpt_content = implode("​", $excerpt_zeo_space); //implode by zerowidthspace
+				endif;
 
-		if (!empty($read_more)):
-			 $color_name = odm_country_manager()->get_current_country().'-color';
-			 $excerpt_words .=  " <a href='".get_permalink($post->ID)." ' class='".$color_name."'>".__($read_more, 'odm').'</a>';
-		endif;
+				$excerpt_words = $excerpt_content.' ...';
+
+				if ($read_more != '') {
+					 $color_name = odm_country_manager()->get_current_country().'-color';
+					 $excerpt_words .=  " (<a href='".get_permalink($post->ID)." ' class='".$color_name."'>".__($read_more, 'odm').'</a>)';
+				}
+				return '<p>' . $excerpt_words . '</p>';
+			endif;
+		}
 
-		return '<p>' . $excerpt_words . '</p>';
+		return;
```

Comment:
@Huyeng nothing is retuned in this case? Maybe we should change it to return ```$translated_content``` in case ```$stripped_content``` is not set. (which I believe is not possible since strip_shortcodes always returns a string , see https://codex.wordpress.org/Function_Reference/strip_shortcodes )